### ISSUE TITLE
Cancelling a pull recycles items

### ISSUE THREAD

**Issue by markgibbons25** [JeffDarchuk/SitecoreSidekick#61]
In 1.5.10, if you pull a root item where you already have some of the tree but not all populated, and then hit Cancel, it recycles a lot of items that shouldn't be.

Here I waited while it skipped past 543 existing items, then pressed cancel and it immediately recycled a lot of items. I don't understand why certain items are being recycled but they definitely exist on the source server.

![image](https://user-images.githubusercontent.com/3212534/97651212-2ce50800-1aa7-11eb-937a-4198ad4b4471.png)

![image](https://user-images.githubusercontent.com/3212534/97651447-c9a7a580-1aa7-11eb-93b7-b2d395bd489b.png)

Also attached the error 

```
{002aeb82-067c-455b-a1be-bfd35a0254b7}
System.Exception: Transaction (Process ID 108) was deadlocked on lock resources with another process and has been chosen as the deadlock victim. Rerun the transaction. ---> System.Data.DataException: Error executing SQL command:  INSERT INTO [ArchivedFields] ( [RowId], [ArchivalId], [VersionId],
          [FieldId], [Value], [Created], [Updated]) 
          SELECT [Id], @archivalId, @versionId, [FieldId], [Value], [Created], [Updated]
          FROM [VersionedFields] 
          WHERE [ItemId] = @itemId and [Language] = @language AND [Version] = @version ---> System.Data.SqlClient.SqlException: Transaction (Process ID 108) was deadlocked on lock resources with another process and has been chosen as the deadlock victim. Rerun the transaction.
   at System.Data.SqlClient.SqlConnection.OnError(SqlException exception, Boolean breakConnection, Action`1 wrapCloseInAction)
   at System.Data.SqlClient.TdsParser.ThrowExceptionAndWarning(TdsParserStateObject stateObj, Boolean callerHasConnectionLock, Boolean asyncClose)
   at System.Data.SqlClient.TdsParser.TryRun(RunBehavior runBehavior, SqlCommand cmdHandler, SqlDataReader dataStream, BulkCopySimpleResultSet bulkCopyHandler, TdsParserStateObject stateObj, Boolean& dataReady)
   at System.Data.SqlClient.SqlCommand.FinishExecuteReader(SqlDataReader ds, RunBehavior runBehavior, String resetOptionsString, Boolean isInternal, Boolean forDescribeParameterEncryption, Boolean shouldCacheForAlwaysEncrypted)
   at System.Data.SqlClient.SqlCommand.RunExecuteReaderTds(CommandBehavior cmdBehavior, RunBehavior runBehavior, Boolean returnStream, Boolean async, Int32 timeout, Task& task, Boolean asyncWrite, Boolean inRetry, SqlDataReader ds, Boolean describeParameterEncryptionRequest)
   at System.Data.SqlClient.SqlCommand.RunExecuteReader(CommandBehavior cmdBehavior, RunBehavior runBehavior, Boolean returnStream, String method, TaskCompletionSource`1 completion, Int32 timeout, Task& task, Boolean& usedCache, Boolean asyncWrite, Boolean inRetry)
   at System.Data.SqlClient.SqlCommand.InternalExecuteNonQuery(TaskCompletionSource`1 completion, String methodName, Boolean sendToPipe, Int32 timeout, Boolean& usedCache, Boolean asyncWrite, Boolean inRetry)
   at System.Data.SqlClient.SqlCommand.ExecuteNonQuery()
   at Sitecore.Data.DataProviders.Sql.DataProviderCommand.ExecuteNonQuery()
   --- End of inner exception stack trace ---
   --- End of inner exception stack trace ---
   at Sitecore.Data.DataProviders.Sql.DataProviderCommand.ExecuteNonQuery()
   at Sitecore.Data.DataProviders.Sql.SqlDataApi.<>c__DisplayClass33_0.<Execute>b__0()
   at Sitecore.Data.DataProviders.NullRetryer.Execute[T](Func`1 action, Action recover)
   at Sitecore.Data.Archiving.SqlArchive.ArchiveVersionedFields(ID itemId, ID archivalId, String language, Int32 version)
   at Sitecore.Data.Archiving.SqlArchive.ArchiveVersionedFields(ID itemId, ID archivalId)
   at Sitecore.Data.Archiving.SqlArchive.ArchiveData(Item item, ID archivalId)
   at Sitecore.Data.Archiving.SqlArchive.<>c__DisplayClass41_0.<DoArchiveItems>b__0(Item item)
   at System.Linq.Enumerable.Any[TSource](IEnumerable`1 source, Func`2 predicate)
   at Sitecore.Data.Archiving.SqlArchive.DoArchiveItems(IEnumerable`1 items, ID archivalId)
   at Sitecore.Data.Archiving.SqlArchive.DoArchiveItems(IEnumerable`1 items, ID archivalId)
   at Sitecore.Data.DataProviders.NullRetryer.Execute[T](Func`1 action, Action recover)
   at Sitecore.Data.Archiving.SqlArchive.ArchiveItem(Item item)
   at SitecoreSidekick.Services.SitecoreDataAccessService.RecycleItem(String id)
   at ScsContentMigrator.Core.ContentItemInstaller.CleanUnwantedLocalItems()
```

**Comment by JeffDarchuk** [comment]
Whoa, that's bad, I'll get that fixed up asap tomorrow

**Comment by JeffDarchuk** [comment]
i've pushed the fix into 1.5.11

The error you get there seems to be triggered from inside Sitecore.  Recycle calls are done using the plain ol SC Item API, I got a few of those as well in my testing.  I don't think it's something to be overly concerned about.  The way it works is it deletes from the leaf nodes down using a single thread so it's very likely to end up deleted from a parent's deletion anyway.  Only thing i can think of to mitigate would be some kind of Item API retry mechanism, might be worth a try.

**Comment by markgibbons25** [comment]
I can confirm this is fixed in 1.5.11
